MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/PolygonSelectToggleActivity.kt

```diff
@@ -0,0 +1,290 @@
+                fillOpacity(.95f))
+
+        // Add a filter to ensure that only "selected" features/polygons/neighborhoods are displayed.
+        neighborhoodPolygonColoredFillLayer.withFilter(eq((get(PROPERTY_SELECTED)), literal(true)))
+        mapboxMap?.style?.addLayerBelow(neighborhoodPolygonColoredFillLayer, "settlement-label")
+
+        // Add LineLayer to outline the various neighborhoods' areas
+        val neighborhoodOutlineLineLayer = LineLayer(LINE_LAYER_ID, NEIGHBORHOOD_POLYGON_SOURCE_ID)
+        neighborhoodOutlineLineLayer.withProperties(
+                lineColor(Color.GRAY),
+                lineWidth(2.2f))
+        mapboxMap?.style?.addLayerBelow(neighborhoodOutlineLineLayer, "settlement-label")
+    }
+
+    /**
+     * This method handles click events for SymbolLayer symbols.
+     * When a SymbolLayer icon is clicked, we adjust that feature to the selected state.
+     * @param screenPoint the point on screen clicked
+     */
+    private fun handleClickIcon(screenPoint: PointF): Boolean {
+        val features = mapboxMap?.queryRenderedFeatures(screenPoint, BASE_NEIGHBORHOOD_FILL_LAYER_ID)
+        if (features?.isEmpty() == false) {
+            val name = features[0].getStringProperty(NEIGHBORHOOD_NAME_PROPERTY)
+            val featureList = featureCollection?.features()
+            for (i in featureList!!.indices) {
+                if (featureList[i].getStringProperty(NEIGHBORHOOD_NAME_PROPERTY) == name) {
+                    if (featureSelectStatus(i)) {
+                        setFeatureSelectState(featureList.get(i), false)
+                    } else {
+                        setSelected(i)
+                    }
+                }
+            }
+            return true
+        } else {
+            return false
+        }
+    }
+
+    /**
+     * Selects the state of a feature
+     *
+     * @param feature the feature to be selected.
+     */
+    private fun setFeatureSelectState(feature: Feature, selectedState: Boolean) {
+        feature.properties()!!.addProperty(PROPERTY_SELECTED, selectedState)
+        refreshSource()
+    }
+
+    /**
+     * Checks whether a Feature's boolean "selected" property is true or false
+     *
+     * @param index the specific Feature's index position in the FeatureCollection's list of Features.
+     * @return true if "selected" is true. False if the boolean property is false.
+     */
+    private fun featureSelectStatus(index: Int): Boolean {
+        featureCollection?.features()!![index].getBooleanProperty(PROPERTY_SELECTED)
+        return if (featureCollection == null) {
+            false
+        } else featureCollection?.features()!![index].getBooleanProperty(PROPERTY_SELECTED)
+    }
+
+    /**
+     * Updates the display of data on the map after the FeatureCollection has been modified
+     */
+    private fun refreshSource() {
+        if (geoJsonSource != null && featureCollection != null) {
```

Comment:
null checking is not needed as in next line you are using `?`

Reply:
removed